project/recursive_sorting.py

```diff
@@ -46,19 +46,28 @@ def merge_sort_in_place(arr, l, r):
 def quick_sort( arr, low, high ):
     if high - low <= 0:
         return arr
-    
+        # base case above, returns array in last position minus zeroeth is less than 0
+        # first rule which is to have a base case
     startingPoint = low
-    for i in range(startingPoint+1, high-1):
-        if arr[i] < arr[startingPoint]:
+    # second rule which is have a path to the base case
+    for i in range(startingPoint+1, high+1):
```

Comment:
I ran this code and it works just fine, Joe! however I'm not sure if this exactly quick sort. you're doing two swaps down there, but not really comparing to a pivot.
that said, I may have misunderstood! or maybe you did like the hungarian dance video?
either way, great work getting this to sort properly.